cgmanifest.json

```diff
@@ -22,55 +22,50 @@
     },
     {
       "Component": {
-        "Type": "other",
-        "Other": {
-          "Name": "cmocka",
-          "Version": "1.1.5-1",
-          "DownloadUrl": "https://gitlab.com/cmocka/cmocka"
+        "Type": "git",
+        "git": {
+          "RepositoryUrl": "https://gitlab.com/cmocka/cmocka",
+          "CommitHash": "f5e2cd77c88d9f792562888d2b70c5a396bfbf7a"
         }
       },
       "DevelopmentDependency": true
     },
     {
       "Component": {
-        "Type": "other",
-        "Other": {
-          "Name": "paho-mqtt",
-          "Version": "1.3.1",
-          "DownloadUrl": "https://github.com/eclipse/paho.mqtt.c"
+        "Type": "git",
+        "git": {
+          "RepositoryUrl": "https://github.com/eclipse/paho.mqtt.c",
```

Comment:
How are we planning to keep the hashes in sync with VCPKG?

Reply:
We have pinned the vcpkg version that we use. These hashes won't change until we do a manual update. At that point, we will update these too.
https://github.com/Azure/azure-sdk-for-c/blob/6395acce257d21df607d92efdffea6d4729e4655/eng/vcpkg-commit.txt#L1

These hashes correspond to the particular version we use.

Reply:
It might be worth adding some validation or automation to keep these in sync as I do think this will likely be forgotten. However I will also say that this is mostly for pulling the license information which rarely if ever changes so it might not be a big deal.

Reply:
Hopefully this issue captures your concern (I am assuming you are referring to keeping vcpkg-commit file and the corresponding commit hashes in cgmanifest):
https://github.com/Azure/azure-sdk-for-c/issues/1364